src/command/mod.rs

```diff
@@ -1,32 +1,90 @@
+use std::time::Duration;
+
+use tokio::sync::oneshot;
+
 use crate::codec::Token;
+use crate::server::Context;
+use crate::storage::StorageCommand;
 
 mod command;
 pub use command::{Command, CommandError};
 
 /// CommandProcessor is responsible for taking a group of tokens, executing them,
 /// and returning the result.
-pub struct CommandProcessor {}
+pub struct CommandProcessor {
+    context: Context,
+}
 
 impl CommandProcessor {
-    pub fn execute_command(&self, command: &Command) -> Vec<Token> {
+    pub fn new(context: Context) -> Self {
+        Self { context }
+    }
+
+    pub async fn execute_command(&self, command: &Command) -> Vec<Token> {
         match command {
             Command::Echo(t) => vec![Token::BulkString(Some(t.clone()))],
             Command::Command => vec![Token::BulkString(Some("ECHO".bytes().collect()))],
+            Command::Get(key) => {
+                let cmd = StorageCommand::Get(key.clone());
+                let (tx, rx) = oneshot::channel();
+                let res = self
+                    .context
+                    .storage_queue
+                    .send_timeout((cmd, tx), Duration::from_millis(1_000))
+                    .await;
+
+                if let Err(_) = res {
+                    return error_response("timeout while sending to storage".to_string());
+                }
+
+                match rx.await {
+                    Ok(Ok(None)) => vec![Token::BulkString(None)],
+                    Ok(Ok(Some(value))) => vec![Token::BulkString(Some(value))],
+                    Ok(Err(_)) => error_response("internal storage error".to_string()),
+                    Err(_) => error_response("no response from storage".to_string()),
```

Comment:
My spidey sense is saying that this function should be split in two: one that handles app logic and returns an enum indicating the response, another that renders that enum into a token to be sent to the client.

Reply:
Quite right, we split it a lot!